app/api/product.py

```
@@ -1,167 +1,262 @@
+                "상품이 추천 시스템에 등록되어 있습니다" if is_indexed else 
+                "POST /products/embedding/generate 를 사용하여 이 상품을 추천 시스템에 등록하세요"
+            ]
         }
         
     except HTTPException:
         raise
     except Exception as e:
-        logger.error(f"상품 임베딩 생성 실패: {e}")
-        raise HTTPException(status_code=500, detail=f"임베딩 생성 실패: {str(e)}") from e
+        logger.error(f"임베딩 검증 실패: {e}")
+        raise HTTPException(status_code=500, detail=f"임베딩 검증 실패: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)